var/spack/repos/builtin/packages/xl/package.py

```diff
@@ -79,3 +62,10 @@ def cxx(self):
             return self.spec.extra_attributes["compilers"]["cxx"]
         msg = "cannot retrieve C compiler [spec is not concrete]"
         assert self.spec.concrete, msg
+
+    @property
+    def fortran(self):
+        if self.spec.external:
+            return self.spec.extra_attributes["compilers"]["fortran"]
+        msg = "cannot retrieve C compiler [spec is not concrete]"
```

Comment:
The message is reporting the wrong language (also the previous C++ message). Also, there's a mismatch between `compilers` and `paths` (I still prefer `compilers` because `paths` seems a bit generic on an external).